test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -857,25 +966,56 @@ def test_cgroup_periodic_update(self):
         self.server.status(JOB, ATTR_o, jid)
         o = j.attributes[ATTR_o]
         if self.paths['cpuacct']:
-            self.hostA.log_match("%s;update_job_usage: CPU usage: 0.000 secs"
-                                 % jid)
-        if self.paths['memory']:
             self.hostA.log_match(
-                "%s;update_job_usage: Memory usage: mem=0b" % jid)
-            if self.swapctl == "true":
-                self.hostA.log_match(
-                    "%s;update_job_usage: Memory usage: vmem=0b" % jid)
+                '%s;update_job_usage: CPU usage: 0.000 secs' % jid)
+        if self.paths['memory']:
+            line = self.hostA.log_match(
+                '%s;update_job_usage: Memory usage: mem=' % jid)
+            match = re.match(r'.*mem=(\d+)kb', line[1])
+            self.assertFalse(match is None)
+            usage = int(match.groups()[0])
+            self.assertTrue(usage < 400000)
+            if self.swapctl == 'true':
+                line = self.hostA.log_match(
+                    '%s;update_job_usage: Memory usage: vmem=' % jid)
+                match = re.match(r'.*vmem=(\d+)kb', line[1])
+                self.assertFalse(match is None)
+                usage = int(match.groups()[0])
+                self.assertTrue(usage < 400000)
         # Allow some time to pass for values to be updated
+        self.logger.info('Waiting for periodic hook to update usage data.')
         time.sleep(5)
         if self.paths['cpuacct']:
             self.hostA.log_match(
-                "%s;update_job_usage: CPU usage: [0-9.]+ secs" % jid,
+                '%s;update_job_usage: CPU usage: [0-9].[0-9]+ secs' % jid,
                 regexp=True)
         if self.paths['memory']:
-            self.hostA.log_match(
-                "%s;update_job_usage: Memory usage: mem=[1-9][0-9]+kb"
-                % jid, regexp=True)
-            self.wait_and_remove_file(filename=o.split(':')[1], mom=self.momA)
+            lines = self.hostA.log_match(
+                '%s;update_job_usage: Memory usage: mem=' % jid,
+                allmatch=True)
+            usage = 0
+            for line in lines:
+                match = re.match(r'.*mem=(\d+)kb', line[1])
+                if not match:
+                    continue
+                usage = int(match.groups()[0])
+                if usage > 400000:
+                    break
+            self.assertTrue(usage > 400000)
+            if self.swapctl == 'true':
+                lines = self.hostA.log_match(
+                    '%s;update_job_usage: Memory usage: vmem=' % jid,
+                    allmatch=True, max_attempts=5)
+                usage = 0
+                for line in lines:
+                    match = re.match(r'.*mem=(\d+)kb', line[1])
```

Comment:
Same as above, use re.search() (and for other calls to re.match() using with .*)

Reply:
Done.